binding/Binding/HandleDictionary.cs

```diff
@@ -170,45 +162,31 @@ internal static void DeregisterHandle (IntPtr handle, SKObject instance)
 			if (instance is ISKSkipObjectRegistration)
 				return;
 
-			instancesLock.EnterWriteLock ();
-			try {
-				var existed = instances.TryGetValue (handle, out var weak);
-				if (existed && (!weak.IsAlive || weak.Target == instance)) {
-					instances.Remove (handle);
-				} else {
+			var existed = instances.TryUpdate (handle, emptyRef, instance.WeakHandle);
+			if (existed) {
+				// TODO: remove from the dictionary
+				//       or, better yet, do this in a single op
```

Comment:
This is the reason I originally switched away from the dictionary. There is no easy way to have an atomic comparison between the key and value. I only want to remove if the key AND value match. 

The reason for this is just because the native side will have created a brand new object at the same location, and we mustn't remove it form the dictionary. We only do that if this is our handle AND our object.